libraries/AP_Mount/AP_Mount_Siyi.cpp

```diff
@@ -717,45 +719,63 @@ bool AP_Mount_Siyi::set_zoom(ZoomType zoom_type, float zoom_value)
                 return false;
             case HardwareModel::A8:
                 // set internal zoom control target
+                _zoom_type = zoom_type;
                 _zoom_mult_target = zoom_mult;
                 return true;
             case HardwareModel::ZR10:
-                return send_zoom_mult(zoom_mult);
+                if (send_zoom_mult(zoom_mult)) {
+                    _zoom_type = zoom_type;
+                    _zoom_mult_target = 0;
+                    return true;
+                }
+                return false;
             }
         }
+        return false;
+    }
     }
 
     // unsupported zoom type
     return false;
 }
 
-// update absolute zoom controller
-// only used for A8 that does not support abs zoom control
+// update zoom controller
 void AP_Mount_Siyi::update_zoom_control()
 {
-    // exit immediately if no target
-    if (!is_positive(_zoom_mult_target)) {
-        return;
-    }
-
-    // limit update rate to 20hz
     const uint32_t now_ms = AP_HAL::millis();
-    if ((now_ms - _last_zoom_control_ms) <= 50) {
-        return;
-    }
-    _last_zoom_control_ms = now_ms;
+    const uint32_t update_diff_ms = now_ms - _last_zoom_control_ms;
 
-    // zoom towards target zoom multiple
-    if (_zoom_mult_target > _zoom_mult + 0.1f) {
-        send_zoom_rate(1);
-    } else if (_zoom_mult_target < _zoom_mult - 0.1f) {
-        send_zoom_rate(-1);
-    } else {
-        send_zoom_rate(0);
-        _zoom_mult_target = 0;
+    switch (_zoom_type) {
+    case ZoomType::RATE:
+        // limit updates to 1hz
+        if (update_diff_ms < 1000) {
```

Comment:
siyi said 2 in recent communications

Reply:
As discussed separately, sending at 1hz is even better than 2hz because slower is better because each time a zoom rate command arrives the gimbal's camera zoom will stutter slightly.  .. as long as we send a command more often that once-every-2seconds.. because if we don't send within 2 seconds the camera zoom will also stop.